book/layout.md

```diff
@@ -201,113 +213,72 @@ BLOCK_ELEMENTS = [
 ]
 ```
 
-We'll use use `BlockLayout` for elements whose children are in that
-list, and `InlineLayout` otherwise. Let's put that logic in a new
-`layout_mode` function:
+We'll use `BlockLayout` for elements with children from that list, and
+`InlineLayout` otherwise. Put that logic in a new `layout_mode`
+function:
 
 ``` {.python}
 def layout_mode(node):
-    has_text = False
-    has_containers = False
-    for child in node.children:
-        if isinstance(child, Text):
-            has_text = True
-        elif child.tag in BLOCK_ELEMENTS:
-            has_containers = True
-        else:
-            has_text = True
-    if has_containers or not has_text:
-        return "block"
-    else:
+    if isinstance(node, Text):
         return "inline"
+    elif node.children:
+        for child in node.children:
+            if child.tag in BLOCK_ELEMENTS:
+                return "block"
+        return "inline"
+    else:
+        return "block"
 ```
 
-`BlockLayout` can now create child layout objects by looping over its
-children, calling `layout_mode` to determine what type of layout
-object to create:
+The cases make sure text nodes get inline layout while empty elements
```

Comment:
Not sure if this sentence is understandable. Is it necessary either? Maybe remove?

Reply:
ditto